parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -50,109 +51,187 @@ class ManifestDataAndPaths(NamedTuple):
         except (KeyError, StopIteration):
             pass
-        manifest = await workspacefs.remote_loader.load_manifest(
+        manifest = await self._remote_loader.load_manifest(
             entry_id, version=version, timestamp=timestamp
         )
-        if manifest.id not in manifest_cache:
-            manifest_cache[manifest.id] = {}
-        if manifest.version not in manifest_cache[manifest.id]:
-            manifest_cache[manifest.id][manifest.version] = (timestamp, timestamp, manifest)
+        if manifest.id not in self._manifest_cache:
+            self._manifest_cache[manifest.id] = {}
+        if manifest.version not in self._manifest_cache[manifest.id]:
+            self._manifest_cache[manifest.id][manifest.version] = CacheEntry(
+                timestamp, timestamp, manifest
+            )
         elif timestamp:
             if (
-                manifest_cache[manifest.id][manifest.version][1] is None
-                or timestamp > manifest_cache[manifest.id][manifest.version][1]
+                self._manifest_cache[manifest.id][manifest.version].late is None
+                or timestamp > self._manifest_cache[manifest.id][manifest.version].late
             ):
-                manifest_cache[manifest.id][manifest.version] = (
-                    manifest_cache[manifest.id][manifest.version][0],
-                    timestamp,
-                    manifest,
+                self._manifest_cache[manifest.id][manifest.version] = CacheEntry(
+                    self._manifest_cache[manifest.id][manifest.version].early, timestamp, manifest
                 )
             if (
-                manifest_cache[manifest.id][manifest.version][0] is None
-                or timestamp < manifest_cache[manifest.id][manifest.version][0]
+                self._manifest_cache[manifest.id][manifest.version].early is None
+                or timestamp < self._manifest_cache[manifest.id][manifest.version].early
             ):
-                manifest_cache[manifest.id][manifest.version] = (
-                    timestamp,
-                    manifest_cache[manifest.id][manifest.version][1],
-                    manifest,
+                self._manifest_cache[manifest.id][manifest.version] = CacheEntry(
+                    timestamp, self._manifest_cache[manifest.id][manifest.version].late, manifest
                 )
         return manifest
 
-    async def _list_versions(entry_id: EntryID):
-        if entry_id in versions_list_cache:
-            return versions_list_cache[entry_id]
-        versions_list_cache[entry_id] = await workspacefs.remote_loader.list_versions(entry_id)
-        return versions_list_cache[entry_id]
 
-    async def _get_past_path(entry_id: EntryID, version=None, timestamp=None) -> FsPath:
+class VersionsListCache:
+    """
+    Caches results of the remote_loader.list_versions calls (working on EntryIDs).
+    """
+
+    def __init__(self, remote_loader):
+        self._versions_list_cache = {}
```

Comment:
How long does this cache live in memory?

Reply:
It is only needed during the list_version function execution, so I added a comment to explain